app/src/main/java/edu/calvin/cs262/cs262d/eventconnect/views/AddEvent.java

```diff
@@ -53,35 +104,85 @@ public boolean onOptionsItemSelected(MenuItem item) {
         //TODO: Enforce a title and description??
         String title = eventTitle.getText().toString();
         String desc = eventDescription.getText().toString();
-        String host = eventHost.getText().toString();
         String loc = eventLocation.getText().toString();
-        String date = eventDate.getText().toString();
+        //String date = eventDate.getText().toString();
+        String host = eventHost.getText().toString();
+        Event event = new Event();
+
         double cost;
         try {cost = Double.parseDouble(eventCost.getText().toString());}
         catch (java.lang.NumberFormatException e) {cost = 0;}
+
         int threshold;
-        int capacity;
+        final int capacity;
+
         //TODO: prevent user from putting illegal threshold and capacity in.
-        try {threshold = (int) Math.floor(Double.parseDouble(eventCost.getText().toString()));}
-        catch (java.lang.NumberFormatException e) {threshold = 1;}
-        try {capacity = (int) Math.floor(Double.parseDouble(eventCapacity.getText().toString()));}
-        catch (java.lang.NumberFormatException e) {capacity = -1;}
+        try {threshold = (int) Math.floor(Double.parseDouble(eventThreshold.getText().toString()));
+            event.setMinThreshold(threshold);
+            }
+        catch (java.lang.NumberFormatException e) {
+            eventThreshold.setError(getString(R.string.error_invalid_number));
+            return;
+        }
+        catch (RuntimeException e){
+            eventThreshold.setError(getString(R.string.error_invalid_MinNumber));
+            return;
+        }
+
+        try {String capacityText = eventCapacity.getText().toString();
+            // if the capacity is not an empty string, turn it into a number
+            if (!capacityText.equals("")){
+                capacity = (int) Math.floor(Double.parseDouble(capacityText));
+                event.setMaxCapacity(capacity);
+            }
+            // other wise set to -1 which mean there is no max capacity
+            else {
+                capacity = -1;
+            }}
+        catch (java.lang.NumberFormatException e) {
+            eventCapacity.setError(getString(R.string.error_invalid_number));
+            return;
+        }
+        catch (RuntimeException e){
+            eventCapacity.setError(getString(R.string.error_invalid_MaxNumber));
+            return;
+        }
 
         //store the data from the UI elements
-        Event event = new Event();
         event.setTitle(title);
         event.setDescription(desc);
-        event.setHost(host);
-        event.setLocation(loc);
-        event.setDate(date);
+
+        try { String hostText = eventHost.getText().toString();
+            if (!hostText.equals("")){
+                event.setHost(host);
+            } else {
+                eventHost.setError(getString(R.string.error_empty_host));
+                return;
+            }
+        }
+        catch (RuntimeException e){}
```

Comment:
I think you should put a comment in here asking future developers to remember this catch block if we ever throw a runtimeException in the Event class for event.setHost(host).